### ISSUE TITLE
the icon of button disappear.

### ISSUE THREAD

**Issue by alieonsido** [linuxdeepin/deepin-screen-recorder#38]
Hi.
As title: 
![image](https://user-images.githubusercontent.com/14144668/82006243-f272e480-9699-11ea-91b8-298257402499.png)

you can see that: left hand side icon of option button is disappeared.

Environment: Linux arch 5.6.11-arch1-1 #1 SMP PREEMPT Wed, 06 May 2020 17:32:37 +0000 x86_64 GNU/Linux
Application Version: up-to-date by Pacman.

How do I deal with these?
Thanks for your reply.

**Comment by glide-the** [comment]
manjaro kde in the replo

**Comment by glide-the** [comment]
manjaro kde in the problem appear

**Comment by uzarubin** [comment]
I'm having the same issue with archlabs.

**Comment by raul-xag** [comment]
same here. I'm using Manjaro XFCE.

By the way, after a system update a couple days ago, both Deepin Screen Shot and Recorder turned into one new app, called Deepin Screen Capture. Is that right?
By the way, I couldn't find any old release of Recorder I could temporarily run. Downloaded some, but they simply don't run for me.

![Screenshot_2020-05-19_20-55-18](https://user-images.githubusercontent.com/49216144/82390266-a49d0a00-9a14-11ea-873b-45b91e276c7b.png)

**Comment by shenqizhi** [comment]
Hi, developer 

Same issue as raul-xag in Manjaro XFCE. Please help to check!

Many Thanks

**Comment by JJK96** [comment]
Same here in i3 on Arch linux

Seems like it is not related to the window manager

**Comment by dotNetDR** [comment]
Same issue

OS: Manjaro 20.0.1 Lysia
Kernel: x86_64 Linux 5.4.40-1-MANJARO
Shell: bash 5.0.16
DE: Xfce4
WM: Xfwm4
WM Theme: Matcha-sea
GTK Theme: Matcha-dark-sea [GTK2]
Icon Theme: Papirus-Dark-Maia
Font: Noto Sans 10
![IMG_2633](https://user-images.githubusercontent.com/926166/82732872-ae8b7980-9d42-11ea-90e1-d696d3bad5e0.JPG)

**Comment by JojiiOfficial** [comment]
Same Issue on Arch/Plasma

**Comment by freeNestor** [comment]
> Same issue
> 
> OS: Manjaro 20.0.1 Lysia
> Kernel: x86_64 Linux 5.4.40-1-MANJARO
> Shell: bash 5.0.16
> DE: Xfce4
> WM: Xfwm4
> WM Theme: Matcha-sea
> GTK Theme: Matcha-dark-sea [GTK2]
> Icon Theme: Papirus-Dark-Maia
> Font: Noto Sans 10
> ![IMG_2633](https://user-images.githubusercontent.com/926166/82732872-ae8b7980-9d42-11ea-90e1-d696d3bad5e0.JPG)

same here, and the font is white hard to see on my desktop.
![WeChat Image_20200607102351](https://user-images.githubusercontent.com/19606302/83958888-91f65200-a8a9-11ea-9bd2-d2d21f5eef18.jpg)
![WeChat Image_20200607102347](https://user-images.githubusercontent.com/19606302/83958891-991d6000-a8a9-11ea-8d18-4fd88d442e23.jpg)

**Comment by gkotian** [comment]
I have the exact same issue (i3 on Arch Linux). It's such a great tool, but it has become so difficult to use :disappointed: 

![WhatsApp Image 2020-06-16 at 22 54 56](https://user-images.githubusercontent.com/1580240/84827251-75ce7f80-b024-11ea-865e-19049210f386.jpeg)

**Comment by hellojukay** [comment]
Manjaro XFCE also.

**Comment by ckji** [comment]
Same issue on Manjaro cinnamon

**Comment by RigoLigoRLC** [comment]
I went through the source code and seems that Deepin has deprecated the old way of reading from Qt resource. Now the code obtains the icon from theme so icons disappeared cause other DEs won't have the Deepin icons. If Plasma user try installing a DDE and apply Deepin icon theme, icons might come back.

I think we should discuss whether we should keep this implementation...

**Comment by RigoLigoRLC** [comment]
[src/widgets/subtoolwidget.cpp](../tree/master/src/widgets/subtoolwidget.cpp) line 139 : 151
```C++
    audioButton->setObjectName("AudioButton");
    audioButton->setText(" ");
    audioButton->setIconSize(TOOL_ICON_SIZE);
//    audioButton->setToolTip(tr("Sound On"));
    installTipHint(audioButton, tr("Sound On"));
//    if (m_themeType == 1) {
//        audioButton->setIcon(QIcon(":/image/newUI/normal/microphone_normal.svg"));
//    }

//    else if (m_themeType == 2) {
//        audioButton->setIcon(QIcon(":/image/newUI/dark/normal/microphone_normal.svg"));
//    }
    audioButton->setIcon(QIcon::fromTheme("microphone_normal"));
```

Seems they wanted to simplify the process of determining dark/light theme by utilizing system theme settings, but didn't expect that this affected the users of other DEs.

**Comment by RigoLigoRLC** [comment]
![image](https://user-images.githubusercontent.com/49363666/86434161-495a7900-bd2f-11ea-8b23-2e12ff4eeee7.png)
Like so.

**Comment by freeNestor** [comment]
So, is the only way fixing it rewriting the code? I would cry.

**Comment by raul-xag** [comment]
**for those who use debian-based distros (Debian, Ubuntu, Mint, etc), I got the .deb files for the old "deepin screenshot" and "deepin screen recorder" from Debian 10. It's working fine on Debian like months ago, before the change of the interface and the fusion of this 2 apps.

I think you guys should try it.
https://drive.google.com/drive/folders/1E8inB9vmCVuAwhvhWRUeWEqo8agj_2Nz?usp=sharing**

Sadly I couldn't find a backup solution for Manjaro.

As we can see, the developer(s) is not giving support nor even attention for this app anymore, so we need backup solutions and now this page is not the place to solve this. I'm not used to talk to Linux communities, but I'll try to report this somehow to the debian-based distros in the near future.
**Manjaro/Arch users (I'm not one anymore) please report this bug to AUR.**

**Comment by raul-xag** [comment]
@RigoLigoRLC 
an idea:
could you send your edited file and specify its path?

**Comment by RigoLigoRLC** [comment]
> @RigoLigoRLC 
> an idea:
> could you send your edited file and specify its path?

I got all my modifications in the screenshot, and the path is exactly what I've linked in the comments. it's a really easy fix in my test.

**Comment by freeNestor** [comment]
I came across this question again today. After researching some materials about Qt setIcon() function with QIcon::fromTheme(), I got the icon back on my manjaro xfce. There's a way don't need to change the source code, you just need copy icons in this repo into your icon theme direcory. Missing icons are all in image directory in this repo, donwload them and copy into your icon theme directory called "actions/size x size". For example: 
~/.icons/Uos-fulldistro-icons/actions/24 > cp /tmp/deepin-screen-recorder/image/newUI/normal/* .
you can copy other icons under newUI into actions, but I suggest to backup your actions directory avoid being overrided by some same name icons.
Good Luck! 

![DeepinScreenshot_select-area_20201101225743](https://user-images.githubusercontent.com/19606302/97806532-f2da5880-1c96-11eb-8210-57724f7654ac.png)

**Comment by RigoLigoRLC** [comment]
> I came across this question again today. After researching some materials about Qt setIcon() function with QIcon::fromTheme(), I got the icon back on my manjaro xfce. There's a way don't need to change the source code, you just need copy icons in this repo into your current icon theme direcory. Missing icons are all in image directory in this repo, donwload them and copy into your current icon theme directory called "actions/size x size". For example(uos is my current icon theme, and icon files are under my home direcoy): 
> ~/.icons/Uos-fulldistro-icons/actions/24 > cp /tmp/deepin-screen-recorder/image/newUI/normal/* .
> you can copy other icons under newUI into actions, but I suggest to backup your actions directory avoid being overrided by some same name icons.
> Good Luck! 
> 
> ![DeepinScreenshot_select-area_20201101225743](https://user-images.githubusercontent.com/19606302/97806532-f2da5880-1c96-11eb-8210-57724f7654ac.png)
> 

Yes, Deepin DE used this as a lazy way of determining dark/light theme (most likely). I see this as a bad solution. It's not portable, not flexible when dealing with cross-environment issues.

**Comment by freeNestor** [comment]
> Yes, Deepin DE used this as a lazy way of determining dark/light theme (most likely). I see this as a bad solution. It's not portable, not flexible when dealing with cross-environment issues.

Yeh, I agree with you. There's a better solution actually that they can package the icons as fallback resources in Qt development, as QIcon::fromTheme() support fallback argument. Maybe they think little of this but user not.

**Comment by thomasesr** [comment]
On Gnome3 under Ubuntu I'm experiencing the same issues.
